File: frontend/src/types/booking.ts
import type { RecurrencePattern } from "./recurrencePattern";

export interface Route {
    departureStation: string;
    destinationStation: string;
}

export interface Schedule {
    travelDate: string;
    departureTime: string;
    arrivalTime: string;
}

export interface SpecialRequest {
    id?: number;
    description: string;
}

export const BOOKING_TYPES = {
    OneOff: "OneOff",
    Recurring: "Recurring"
} as const;

export type BookingType = typeof BOOKING_TYPES[keyof typeof BOOKING_TYPES];

export interface Booking {
    id: number;
    bookingReference: string;
    seatNumber: string;
    ticketPrice: number;
    bookingType: BookingType;
    recurrencePattern: RecurrencePattern;
    recurrenceEndDate: string;
    route: Route;
    schedule: Schedule;
    specialRequests: SpecialRequest[];
}

File: backend/TrainTicket.API/Models/Booking.cs
namespace TrainTicket.API.Models;

public class Booking
{
    public int Id { get; set; }

    public string BookingReference { get; set; } = "";

    public string SeatNumber { get; set; } = "";

    public decimal TicketPrice { get; set; }

    public BookingType BookingType { get; set; }


    public int RouteId { get; set; }

    public Route Route { get; set; } = null!;


    public int ScheduleId { get; set; }

    public Schedule Schedule { get; set; } = null!;


    public List<SpecialRequest> SpecialRequests { get; set; } = new();
}